src/battle-scene.ts

```diff
@@ -1233,12 +1233,19 @@ export default class BattleScene extends SceneBase {
       newDouble = !!double;
     }
 
-    if (Overrides.BATTLE_TYPE_OVERRIDE === "double") {
-      newDouble = true;
-    }
-    /* Override battles into single only if not fighting with trainers */
-    if (newBattleType !== BattleType.TRAINER && Overrides.BATTLE_TYPE_OVERRIDE === "single") {
-      newDouble = false;
+    if (!isNullOrUndefined(Overrides.BATTLE_TYPE_OVERRIDE)) {
+      if (Overrides.BATTLE_TYPE_OVERRIDE === "double"
+        || (Overrides.BATTLE_TYPE_OVERRIDE === "odd-doubles" && (newWaveIndex % 2))
+        || (Overrides.BATTLE_TYPE_OVERRIDE === "even-doubles" && !(newWaveIndex % 2))
+      ) {
+        newDouble = true;
+      } else if (newBattleType !== BattleType.TRAINER) {
```

Comment:
```suggestion
      } else if (newBattleType !== BattleType.TRAINER && Overrides.BATTLE_TYPE_OVERRIDE === "single") {
```
I think I'd like to keep this check for future proofing in case a new battle type override is added in the future. Also makes it easier to understand